pheme/templatetags/charts/treemap.py

```diff
@@ -0,0 +1,304 @@
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+# Squarified Treemap Layout
+# Implements algorithm from Bruls, Huizing, van Wijk, "Squarified Treemaps"
+#   (but not using their pseudocode)
+
+# It is a simplified, adapted version of squarify:
+# https://github.com/laserson/squarify
+# For more information see:
+# https://www.win.tue.nl/~vanwijk/stm.pdf
+# pylint: disable=C0103
+import math as m
+import numbers
+from dataclasses import dataclass
+from typing import Dict, List
+
+from django.utils.safestring import mark_safe
+
+from pheme.templatetags.charts import _severity_class_colors, register
+
+__ELEMENT_TEMPLATE = """
+<g>
+    <rect x="{x}" y="{y}" width="{width}" height="{height}" fill="{color}" stroke="{border_color}" strokeWidth="1" />
+    <text x="{label_x}" y="{label_y}" width="{width}" height="{height}" dominant-baseline="central">{label}</text>
+</g>
+"""
+
+__TEMPLATE = """
+<svg version="1.1"
+     baseProfile="full"
+     width="{width}" height="{height}"
+     viewBox="0 0 {width} {height}"
+     xmlns="http://www.w3.org/2000/svg">
+     {rects}
+</svg>
+"""
+
+
+@dataclass
+class Rect:
+    x: float
+    y: float
+    dx: float
+    dy: float
+
+
+def __create_rectangle(x, y, dx, dy) -> Rect:
+    # add spacing of 1px
+    if dx > 2:
+        x += 1
+        dx -= 2
+    if dy > 2:
+        y += 1
+        dy -= 2
+    return Rect(x, y, dx, dy)
+
+
+def __layoutrow(sizes, x, y, _, dy) -> List[Rect]:
+    covered_area = sum(sizes)
+    width = covered_area / dy
+    rects = []
+    for size in sizes:
+        rects.append(__create_rectangle(x, y, width, size / width))
+        y += size / width
+    return rects
+
+
+def __layoutcol(sizes, x, y, dx, _) -> List[Rect]:
+    covered_area = sum(sizes)
+    height = covered_area / dx
+    rects = []
+    for size in sizes:
+        rects.append(__create_rectangle(x, y, size / height, height))
+        x += size / height
+    return rects
+
+
+def __layout(sizes, x, y, dx, dy) -> List[Rect]:
+    if dx >= dy:
+        return __layoutrow(sizes, x, y, dx, dy)
+    return __layoutcol(sizes, x, y, dx, dy)
+
+
+def __leftoverrow(sizes, x, y, dx, dy):
+    covered_area = sum(sizes)
+    width = covered_area / dy
+    leftover_x = x + width
+    leftover_y = y
+    leftover_dx = dx - width
+    leftover_dy = dy
+    return leftover_x, leftover_y, leftover_dx, leftover_dy
+
+
+def __leftovercol(sizes, x, y, dx, dy):
+    covered_area = sum(sizes)
+    height = covered_area / dx
+    leftover_x = x
+    leftover_y = y + height
+    leftover_dx = dx
+    leftover_dy = dy - height
+    return leftover_x, leftover_y, leftover_dx, leftover_dy
+
+
+def __leftover(sizes, x, y, dx, dy):
+    if dx >= dy:
+        return __leftoverrow(sizes, x, y, dx, dy)
+    return __leftovercol(sizes, x, y, dx, dy)
```

Comment:
```suggestion
def __leftovercol(covered_area, x, y, dx, dy):
    height = covered_area / dx
    leftover_x = x
    leftover_y = y + height
    leftover_dx = dx
    leftover_dy = dy - height
    return leftover_x, leftover_y, leftover_dx, leftover_dy


def __leftover(sizes, rect: Rect):
    covered_area = sum(sizes)
    if rect.dx >=rect. dy:
        return __leftoverrow(covered_area, rect)
    return __leftovercol(covered_area, rect)
```

maybe also something like this?